src/customElements/applicationGuidance/Wrapper/index.ts

```diff
@@ -0,0 +1,351 @@
+    ApplicationGuidanceAnchor,
+    ApplicationGuidanceScopeIsolation,
+    ApplicationGuidanceScope,
+} from '../types';
+import './fab';
+import './popover';
+import './interactiveAnchor';
+import ApplicationGuidanceApi from '../api';
+
+const createSvgRect = (rect: ApplicationGuidanceAnchorRect) => svg`
+    <rect
+        x="${rect.left}px"
+        y="${rect.top}px"
+        width="${rect.width}px"
+        height="${rect.height}px"
+        rx="4px"
+        fill="black"
+    ></rect>
+`;
+
+@customElement('app-guide-wrapper')
+export class ApplicationGuidanceWrapper extends LitElement {
+    @property({ type: String, attribute: 'scope' })
+    scope = '';
+
+    @property({ type: String, attribute: 'isolation' })
+    isolation: ApplicationGuidanceScopeIsolation = 'none';
+
+    @property({ type: Array, attribute: false })
+    scopes: ApplicationGuidanceScope[] = [];
+
+    @property({ type: Array, attribute: false })
+    anchors: ApplicationGuidanceAnchor[] = [];
+
+    @property({ type: String, attribute: false })
+    activeAnchorId: string | null = null;
+
+    @property({ type: Boolean, attribute: false })
+    isActive: boolean = false;
+
+    @property({ type: Array, attribute: false })
+    quickFacts: ApplicationGuidanceQuickFact[] = [];
+
+    @property({ type: Boolean, attribute: false })
+    private isFetchingActiveQuickFact: boolean = false;
+
+    @property({ type: String, attribute: 'azure-ad-app-id' })
+    clientId: string;
+
+    private api: ApplicationGuidanceApi;
+
+    static get styles() {
+        return css`
+            .overlay {
+                position: fixed;
+                top: 0;
+                left: 0;
+                right: 0;
+                bottom: 0;
+                z-index: 99999999999999;
+                pointer-events: none;
+                opacity: 0;
+                transition: opacity 0.2s;
+            }
+
+            .overlay.active {
+                opacity: 1;
+                pointer-events: all;
+            }
+
+            .anchors {
+                position: fixed;
+                width: 100vw;
+                height: 100vh;
+                top: 0;
+                left: 0;
+            }
+        `;
+    }
+
+    constructor() {
+        super();
+        window.addEventListener('message', this.handleMessage);
+        this.api = new ApplicationGuidanceApi(this.clientId);
+    }
+
+    updated(changedProperties: Map<string, any>) {
+        if (changedProperties.has('scope') || changedProperties.has('scopes')) {
+            this.fetchQuickFacts(this.getScopePath());
+        }
+
+        if (changedProperties.has('activeAnchorId') && this.activeAnchorId) {
+            this.fetchUpdatedQuickFact(this.getScopePath(), this.activeAnchorId);
+        }
+
+        if (!this.isActive && changedProperties.has('isActive')) {
+            this.scopes = [];
+        }
+    }
+
+    disconnectedCallback() {
+        window.removeEventListener('message', this.handleMessage);
```

Comment:
**Other CustomElements that subscribes to message should also implement this.**
*note*
subscription is attached in constructor and not connectedCallback, this might cause issues when react updates virtual dom (not listening for messages after reinserted).